vcs/src/main/java/org/openjdk/skara/vcs/git/GitRepository.java

```diff
@@ -997,6 +1001,15 @@ public void rebase(Hash hash, String committerName, String committerEmail) throw
         }
     }
 
+    public boolean isRemergeDiffEmpty(Hash hash) throws IOException {
+        try (var p = Process.capture("git", "show", "--remerge-diff", "--format=%b", hash.hex())
+                .workdir(dir)
+                .environ(currentEnv)
+                .execute()) {
+            return String.join("", await(p).stdout()).isEmpty();
+        }
+    }
+
```

Comment:
Maybe it's better to add an interface in `ReadOnlyRepository`

Reply:
If we add this method to `ReadOnlyRepository`, then we'll need to provide some implementation of it for `HgRepository`, which is a subclass of `ReadOnlyRepository`. Not only does `HgRepository` need no such functionality (AFAIK, PRs are not modelled to work in Mercurial repos), a Mercurial implementation will likely be more involved than that of Git.

Reply:
Yes, for `HgRepository`,  you just need to put `throw new UnsupportedOperationException();` in the implementation

Reply:
Here's what I'm going to do:

  - pull up new `Commits commits(List<Hash> reachableFrom, List<Hash> unreachableFrom)` to `ReadOnlyRepository`,
  - pull up `boolean isRemergeDiffEmpty(Hash mergeCommitHash)` to `Repository`.

`isRemergeDiffEmpty` is not a good fit for `ReadOnlyRepository` as far as I understand it. Additionally, if in the future we decide to implement `--remerge-diff` manually, for whatever reason, it'll be even less fit.